app/controllers/hq/partners_controller.rb

```diff
@@ -5,4 +5,48 @@ class Hq::PartnersController < ApplicationController
   def index
     @partners= Partner.all
   end
+
+  def edit
+    load_partner
+    load_associations
+  end
+
+  def update
+    load_partner
+    load_associations
+    build_partner
+    save_partner or render 'edit'
+  end
+
+private
+  def load_partner
+    @partner = Partner.find(params[:id])
+  end
+
+  def load_associations
+    @provinces = Province.all
+    @statuses = Status.all
+  end
+
+  def build_partner
+    @partner ||= Partner.build
+    @partner.attributes = partner_params
+  end
+
+  def save_partner
+    if @partner.save
+      redirect_to hq_partner_url(@partner)
+    end
+  end
+
+  def partner_params
+    partner_params = params[:partner]
+    if partner_params
+      partner_params.permit(:name, :region, :contact_details, :province_id,
+                            :status_id, :start_date)
+    else
+      {}
```

Comment:
Why would we want the controller to carry on with an empty params hash when one should be present? i.e. Why this instead of `params.require(:partner).permit(...)`?